modules/wpml/wpml-config.php

```diff
@@ -173,34 +190,18 @@ public function get_files() {
 	}
 
 	/**
-	 * Adds custom fields to the list of metas to copy when creating a new translation.
+	 * Adds post metas to the list of metas to copy when creating a new translation.
 	 *
 	 * @since 1.0
 	 *
-	 * @param string[] $metas The list of custom fields to copy or synchronize.
+	 * @param string[] $metas The list of post metas to copy or synchronize.
 	 * @param bool     $sync  True for sync, false for copy.
-	 * @return string[] The list of custom fields to copy or synchronize.
+	 * @return string[] The list of post metas to copy or synchronize.
+	 *
+	 * @phpstan-param array<non-falsy-string> $metas
 	 */
 	public function copy_post_metas( $metas, $sync ) {
-		foreach ( $this->xmls as $xml ) {
-			$cfs = $xml->xpath( 'custom-fields/custom-field' );
-
-			if ( ! is_array( $cfs ) ) {
-				continue;
-			}
-
-			foreach ( $cfs as $cf ) {
-				$action = $this->get_field_attribute( $cf, 'action' );
-
-				if ( 'copy' === $action || ( ! $sync && in_array( $action, array( 'translate', 'copy-once' ), true ) ) ) {
-					$metas[] = (string) $cf;
-				} else {
-					$metas = array_diff( $metas, array( (string) $cf ) );
-				}
-			}
-		}
-
-		return $metas;
+		return $this->filter_metas_to_copy( (array) $metas, 'custom-fields/custom-field', (bool) $sync );
```

Comment:
Smart move. 
Do you think it's worth to merge the 2 methods `copy_post_metas()` and copy_term_metas()` in `filter_metas_to_copy()` removing the 2nd param and replacing it by something close to:
`$xpath = current_filter() ===  'pll_copy_post_metas' ? 'custom-fields/custom-field' : 'custom-term-fields/custom-term-field'

Reply:
It can be done but I'm not a big fan of it, I prefer to inject the path as a parameter instead of sniffing the current filter.